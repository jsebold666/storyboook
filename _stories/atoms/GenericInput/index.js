import React from 'react';
import { text, selectV2 as select, boolean, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import GenericInput from '../../../components/atoms/GenericInput';

const component = () => (
  <GenericInput onChange={action('changed')} value={text('value', 'GenericInput content')} />
);

export default [readme, component];
