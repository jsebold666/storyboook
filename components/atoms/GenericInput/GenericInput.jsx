import React, { Component } from 'react';
import styled from 'styled-components';
import InputMask from 'react-input-mask';
import PropTypes from 'prop-types';

export default class GenericInput extends Component {
  constructor(props) {
    super();

    this.genericInput = null;

    this.handleRef = this.handleRef.bind(this);
    this.onBlurInputHandler = this.onBlurInputHandler.bind(this);
    this.onFocusInputHandler = this.onFocusInputHandler.bind(this);
    this.onChangeInputHandler = this.onChangeInputHandler.bind(this);
    this.onKeyPressInputHandler = this.onKeyPressInputHandler.bind(this);
  }

  handleRef(input) {
    const { inputRef } = this.props;

    if (inputRef) {
      inputRef(input);
    }
  }

  onChangeInputHandler(ev) {
    const { onChange } = this.props;
    if (onChange) {
      onChange(ev);
    }
  }

  onFocusInputHandler(ev) {
    const { onFocus } = this.props;

    if (onFocus) {
      onFocus(ev);
    }
  }

  onBlurInputHandler(ev) {
    const { onBlur } = this.props;

    if (onBlur) {
      onBlur(ev);
    }
  }

  onKeyPressInputHandler(ev) {
    const { onKeyPress, onEnter } = this.props;

    if (onKeyPress) {
      onKeyPress(ev);
    }

    if (ev.key === 'Enter' && onEnter) {
      onEnter(ev);
    }
  }

  render() {
    const { name, type, className, placeholder, disabled, mask } = this.props;

    return (
      <InputMask
        inputRef={this.handleRef}
        className={className}
        name={name}
        type={type}
        placeholder={placeholder}
        onChange={this.onChangeInputHandler}
        onKeyPress={this.onKeyPressInputHandler}
        onBlur={this.onBlurInputHandler}
        onFocus={this.onFocusInputHandler}
        disabled={disabled}
        mask={mask}
        value={this.props.value}
      />
    );
  }
}

GenericInput.defaultProps = {
  name: 'genericInput',
  type: 'text',
  placeholder: '',
  disabled: false,
  className: ''
};

GenericInput.propTypes = {
  /* input props */
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  mask: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.instanceOf(Date)]),

  /* input events */
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onKeyPress: PropTypes.func,
  onFocus: PropTypes.func,
  onEnter: PropTypes.func
};
