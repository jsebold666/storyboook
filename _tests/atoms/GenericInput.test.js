/* globals mount */
import React from 'react';
import GenericInput from '../../components/atoms/GenericInput';

describe('Expect <GenericInput>', () => {
  it('to render', () => {
    const props = {
      value: 'This is text',
      onChange: () => {}
    };

    const wrapper = mount(<GenericInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('to render properly with empty', () => {
    const props = {
      value: '',
      onChange: () => {}
    };

    const wrapper = mount(<GenericInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
