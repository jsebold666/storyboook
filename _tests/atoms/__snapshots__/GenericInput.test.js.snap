// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <GenericInput> to render 1`] = `
<GenericInput
  className=""
  disabled={false}
  name="genericInput"
  onChange={[Function]}
  placeholder=""
  type="text"
  value="This is text"
>
  <InputElement
    className=""
    disabled={false}
    inputRef={[Function]}
    name="genericInput"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    type="text"
    value="This is text"
  >
    <input
      className=""
      disabled={false}
      name="genericInput"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      placeholder=""
      type="text"
      value="This is text"
    />
  </InputElement>
</GenericInput>
`;

exports[`Expect <GenericInput> to render properly with empty 1`] = `
<GenericInput
  className=""
  disabled={false}
  name="genericInput"
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
>
  <InputElement
    className=""
    disabled={false}
    inputRef={[Function]}
    name="genericInput"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    type="text"
    value=""
  >
    <input
      className=""
      disabled={false}
      name="genericInput"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </InputElement>
</GenericInput>
`;
